### ISSUE TITLE
email address not supporting international domain names

### ISSUE THREAD

**Issue by Sasaxxxx** [dani-garcia/vaultwarden#3509]
### Subject of the issue
Problem starts when using international domain names. fx. xxxx@xn--gr-wla37a.com (xxxx@gršć.com)
When user account is created using xxxx@gršć.com (in version of VaultWarden 1.25.2), now VaultWarden v- 1.28.1 do not support this email address. 
When this email address is typed in the VaultWarden Web Vault, Error message is shown: " Input is not an email address." 

### Deployment environment

Error message: "Generation Failed"

* vaultwarden version: 1. 28.1

* Install method: Docker on Synology server

* Clients used: 
* WebVault, desktop, IOS

* Reverse proxy and version: 

* MySQL/MariaDB or PostgreSQL version: 

* Other relevant details:

### Steps to reproduce

1. account created with username/email: xxxx@gršć.com (v.1.25.1)
2.. version updated to 1.28.1
3. IOS app updated, desktop app updated, edge extension updated (since release of 1.25.1)
4. now when user name/email: xxxx@@gršć.com is used, it is been converted to xxxx@xn--gr-wla37a.com causing error message in WebVault: " Input is not an email address." or in case of App's, no login possible. 

### Expected behaviour

1. when xxxx@gršć.com international domain name is used to create account, domain in account name should not be automatically converted to xxxx@xn--gr-wla37a.com.
2.  

### Actual behaviour

1. LogIn on VaultWarden Web Vault. Email: xxxx@@gršć.com
2.  error message: "Input is not an email address."

**Comment by BlackDex** [comment]
Is the client converting this before it is sending this?

**Comment by Sasaxxxx** [comment]
it was working for over a year. Today I find out I could not log in on IOS App. Updated VaultWarden form 1.25.1 to 1.28.1 with results as reported.

**Comment by Sasaxxxx** [comment]
I can see now following:

Actual behaviour:
LogIn on VaultWarden Web Vault. Email: xxxx@@gršć.com
request for MasterPasword:
underneeth of the logIn button, stated: "Logging in as xxxx@xn--gr-wla37a.com"

so i guess it will try to login as user xxxx@xn--gr-wla37a.com (which is not a user xxxx@@gršć.com). can system tell a difference and convert domains in right way?

**Comment by BlackDex** [comment]
Vaultwarden doesn't convert the address. So it could be that the new clients convert them before sending it to the server.

**Comment by Sasaxxxx** [comment]
is VaultWarden Web Vault server part?

**Comment by BlackDex** [comment]
No, that is client. But it could be that Bitwarden changed something. I'm not sure.

**Comment by BlackDex** [comment]
On a quick look, I'm afraid we can't fix this on our side that easily.
It looks like the web-vault doesn't seem to support this anymore for some reason.

On the web-vault I get a message that it needs to be a valid email address.
It seems to be not working anymore since v1.26.0 already (Which uses v2022.12.0 as web-vault).

**Comment by BlackDex** [comment]
I also think we can't fix this on the server side.
The reason is, the username (mail address) is linked to the hashed password and master-key.
Changing the mail-address, or trying to match the password-hash on the unicode domain will not work.

I suggest to revert back to v1.25.2, change all mail address to use the punycode address via the web-vault and upgrade after all the users have changed it.

It's not something we can fix unfortunately.
